File: scripts/fix-paths.py
#!/usr/bin/env python3
"""
Fix image paths in all Zatiq components
Converts: ../../assets/image/... → /assets/...
"""

import os
import re
from pathlib import Path

def fix_image_paths(file_path):
    """Fix image import paths in a TypeScript/TSX file"""
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    original_content = content
    
    # Pattern 1: import something from '../../assets/image/...'
    content = re.sub(
        r"import\s+(\w+)\s+from\s+['\"]\.\.\/\.\.\/assets\/image\/([^'\"]+)['\"];?",
        r"const \1 = '/assets/\2';",
        content
    )
    
    # Pattern 2: import something from '../../assets/...'  (without image folder)
    content = re.sub(
        r"import\s+(\w+)\s+from\s+['\"]\.\.\/\.\.\/assets\/([^'\"]+)['\"];?",
        r"const \1 = '/assets/\2';",
        content
    )
    
    # Pattern 3: Fix any malformed paths like '/assets/... with missing quote
    content = re.sub(
        r"from\s+'/assets/([^'\"]+)\"",
        r"from '/assets/\1'",
        content
    )
    content = re.sub(
        r'from\s+"/assets/([^\'\"]+)\'',
        r'from "/assets/\1"',
        content
    )
    
    # Pattern 4: Direct string references
    content = re.sub(
        r"['\"]\.\.\/\.\.\/assets\/image\/([^'\"]+)['\"]",
        r"'/assets/\1'",
        content
    )
    
    # Pattern 5: Fix already half-fixed imports
    content = re.sub(
        r"import\s+(\w+)\s+from\s+'/assets/([^'\"]+)['\"];?",
        r"const \1 = '/assets/\2';",
        content
    )
    
    content = re.sub(
        r'import\s+(\w+)\s+from\s+"/assets/([^\'\"]+)[\'"];?',
        r'const \1 = "/assets/\2";',
        content
    )
    
    # Only write if changed
    if content != original_content:
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(content)
        return True
    return False

def main():
    """Process all component files"""
    base_path = Path("src/components/zatiq")
    
    if not base_path.exists():
        print(f"❌ Path not found: {base_path}")
        return
    
    print("🔧 Fixing image paths in components...")
    print()
    
    fixed_count = 0
    total_count = 0
    
    for tsx_file in base_path.rglob("*.tsx"):
        total_count += 1
        if fix_image_paths(tsx_file):
            fixed_count += 1
            print(f"✅ Fixed: {tsx_file.relative_to('src/components/zatiq')}")
    
    print()
    print(f"📊 Summary:")
    print(f"   Total files: {total_count}")
    print(f"   Fixed files: {fixed_count}")
    print(f"   Unchanged: {total_count - fixed_count}")
    print()
    print("✨ Done! Restart dev server to see changes.")

if __name__ == "__main__":
    main()

File: scripts/fix-paths.js
import fs from "fs";
import path from "path";
import { fileURLToPath } from "url";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

function fixImagePaths(filePath) {
  let content = fs.readFileSync(filePath, "utf8");
  const original = content;

  // Pattern 1: import something from '../../assets/image/...'
  content = content.replace(
    /import\s+(\w+)\s+from\s+['"]\.\.\/\.\.\/assets\/image\/([^'"]+)['"]/g,
    "const $1 = '/assets/$2'"
  );

  // Pattern 2: import something from '../../assets/...'
  content = content.replace(
    /import\s+(\w+)\s+from\s+['"]\.\.\/\.\.\/assets\/([^'"]+)['"]/g,
    "const $1 = '/assets/$2'"
  );

  // Pattern 3: Fix malformed imports like: from '/assets/...'
  content = content.replace(
    /import\s+(\w+)\s+from\s+['"]\/assets\/([^'"]+)['"]/g,
    "const $1 = '/assets/$2'"
  );

  // Pattern 4: Direct string paths
  content = content.replace(
    /['"]\.\.\/\.\.\/assets\/image\/([^'"]+)['"]/g,
    "'/assets/$1'"
  );

  content = content.replace(
    /['"]\.\.\/\.\.\/assets\/([^'"]+)['"]/g,
    "'/assets/$1'"
  );

  if (content !== original) {
    fs.writeFileSync(filePath, content, "utf8");
    return true;
  }
  return false;
}

function walkDir(dir) {
  const files = fs.readdirSync(dir);
  let fixed = 0;
  let total = 0;

  files.forEach((file) => {
    const filePath = path.join(dir, file);
    const stat = fs.statSync(filePath);

    if (stat.isDirectory()) {
      const result = walkDir(filePath);
      fixed += result.fixed;
      total += result.total;
    } else if (file.endsWith(".tsx") || file.endsWith(".ts")) {
      total++;
      if (fixImagePaths(filePath)) {
        fixed++;
        console.log(`✅ Fixed: ${filePath.replace(/\\/g, "/")}`);
      }
    }
  });

  return { fixed, total };
}

console.log("🔧 Fixing image paths in components...\n");

const result = walkDir("src/components/zatiq");

console.log(`\n📊 Summary:`);
console.log(`   Total files: ${result.total}`);
console.log(`   Fixed files: ${result.fixed}`);
console.log(`   Unchanged: ${result.total - result.fixed}`);
console.log("\n✨ Done! Restart dev server to see changes.");
